spec/javascripts/stimulus/tabs_and_panels_controller_test.js

```diff
@@ -11,122 +11,143 @@ describe('TabsAndPanelsController', () => {
+
+    const event = new Event("orderCycleSelected")
+    window.dispatchEvent(event);
+
+    expect(peakTab.classList.contains("selected")).toBe(true)
+    expect(kaTab.classList.contains("selected")).toBe(false)
+    expect(booTab.classList.contains("selected")).toBe(false)
+
+    expect(peakPanel.style.display).toBe("block")
+    expect(kaPanel.style.display).toBe("none")
+    expect(booPanel.style.display).toBe("none")
+  })
+
+  describe("when valid anchor is specified in the url", () => {
+    const oldWindowLocation = window.location
+    beforeAll(() => {
+      Object.defineProperty(window, "location", {
+        value: new URL("http://example.com/#boo_panel"),
+        configurable: true,
       })
     })
+    afterAll(() => {
+      delete window.location
+      window.location = oldWindowLocation
+    })
 
-    describe('when anchor is specified in the url', () => {
-      const { location } = window;
-      const mockLocationToString = (panel) => {
-        // Mocking window.location.toString() 
-        const url = `http://localhost:3000/admin/enterprises/great-shop/edit#/${panel}`
-        const mockedToString = jest.fn()
-        mockedToString.mockImplementation(() => (url))
-
-        delete window.location 
-        window.location = {
-          toString: mockedToString
-        } 
-      }
-
-      beforeAll(() => {
-        mockLocationToString('ka_panel')
-      })
-
-      afterAll(() => {
-        // cleaning up
-        window.location = location
-      })
-
-      it('displays the panel associated with the anchor', () => {
-        const kaPanel = document.getElementById('ka_panel');
-
-        expect(kaPanel.style.display).toBe('block');
-      })
-
-      it('selects the tab entry associated with the anchor', () => {
-        const ka = document.getElementById('ka');
-
-        expect(ka.classList.contains('selected')).toBe(true);
+    it("#activateFromWindowLocationOrDefaultPanelTarget show panel based on anchor", () => {
+      const peakTab = document.getElementById("peek_tab")
+      const kaTab = document.getElementById("ka_tab")
+      const booTab = document.getElementById("boo_tab")
+      const peakPanel = document.getElementById("peek_panel")
+      const kaPanel = document.getElementById("ka_panel")
+      const booPanel = document.getElementById("boo_panel")
+
+      expect(peakTab.classList.contains("selected")).toBe(false)
+      expect(kaTab.classList.contains("selected")).toBe(false)
+      expect(booTab.classList.contains("selected")).toBe(true)
+
+      expect(peakPanel.style.display).toBe("none")
+      expect(kaPanel.style.display).toBe("none")
+      expect(booPanel.style.display).toBe("block")
+    })
+  })
+
+  describe("when non valid anchor is specified in the url", () => {
+    const oldWindowLocation = window.location
+    beforeAll(() => {
+      Object.defineProperty(window, "location", {
+        value: new URL("http://example.com/#non_valid_panel"),
+        configurable: true,
```

Comment:
It's a lot cleaner than the previous solution !

Reply:
Yea, no need to use mock